### PR TITLE
configuration.py: add REMOTE_AUTH_ENABLED and REMOTE_AUTH_BACKEND

### PR DESCRIPTION
**Pull request by herbetom**
Related Issue:

#324 

## New Behavior

add the two config variables needed for a temporary fix of the ldap issue
...

## Contrast to Current Behavior

Currently it is needed to override the configuration.py manually which isn't always easy

## Discussion: Benefits and Drawbacks

additioal vars got added by netbox. I want to configure those.

## Changes to the Wiki

https://github.com/netbox-community/netbox-docker/wiki/LDAP

remove the `Netbox v2.9 and above` note or refernce the possibility to use docker environment vars.

## Proposed Release Note Entry

...

## Double Check

* [x] I have read the comments and followed the PR template.
* [x] I have explained my PR according to the information in the comments.
* [x] My PR targets the `develop` branch.

### CODE REVIEW

**Comment by herbetom** [comment]
I don't think that the failed Checks have something to do with my PR. The Check for the latest commit c744603a8557144bf3b170a10a06bc7dbfa88655 in the [develop](https://github.com/netbox-community/netbox-docker/commits/develop) Branch also failed (which my PR is based on).

But i noticed that there is with #310 a PR which also tries to add (among others) those two vars. 

Since our netbox is currently not working, because we can't login, I would appreciate it if something could happen regarding the addition of these vars.

**Comment by johanfleury** [comment]
> Since our netbox is currently not working, because we can't login I would appreciate it if something could happen regarding the addition of these vars.

How much did you pay the maintainer for this?

Also, compared to #310, your PR missing important variables (e.g. `REMOTE_AUTH_HEADER`, `REMOTE_AUTH_AUTO_CREATE_USER`, and `REMOTE_AUTH_DEFAULT_GROUPS`), that will IMO be required in a majority of case.

**Comment by herbetom** [comment]
> How much did you pay the maintainer for this?

Nothing. It's a setup which doesn't use docker-compose. I certaintly could make it work, but i would like a permanet solution (merging of this or (since more options) #310). I just don't see the point in creating a workaround which has to be removed afterwards. 
But since nothing is happening i will have to implement that workaround. 

Since #310 wasn't linked to #324, i wasn't aware of #310. So i just added the vars relevant for #324 to hopefully get this merged as fast as possible. 

It wouldn't be a problem for me to add the other vars or to close this PR if #310 gets merged. But this would require a maintainer (#334) to provide a way forward. @tobiasge @cimnine

**Comment by herbetom** [comment]
I added the missing REMOTE_AUTH_* Options in this PR. But if #310 gets edited i would recommend merging that (as it came first).

File: configuration/configuration.py [hunk 1]
```diff
@@ -187,6 +187,14 @@ def read_secret(secret_name):
 # prefer IPv4 instead.
 PREFER_IPV4 = os.environ.get('PREFER_IPV4', 'False').lower() == 'true'
 
+# Remote authentication support
+REMOTE_AUTH_ENABLED = os.environ.get('REMOTE_AUTH_ENABLED', 'False').lower() == 'true'
+REMOTE_AUTH_BACKEND = os.environ.get('REMOTE_AUTH_BACKEND', 'netbox.authentication.RemoteUserBackend')
+REMOTE_AUTH_HEADER = os.environ.get('REMOTE_AUTH_HEADER', 'HTTP_REMOTE_USER')
+REMOTE_AUTH_AUTO_CREATE_USER = os.environ.get('REMOTE_AUTH_AUTO_CREATE_USER', 'True').lower() == 'true'
+REMOTE_AUTH_DEFAULT_GROUPS = list(filter(None, os.environ.get('REMOTE_AUTH_DEFAULT_GROUPS', '').split(' ')))
+REMOTE_AUTH_DEFAULT_PERMISSIONS = ast.literal_eval(os.environ.get('REMOTE_AUTH_DEFAULT_PERMISSIONS', '{}'))
```
Comment (cimnine) [inline configuration/configuration.py]: I despise an `eval` in the configuration code. Other than that the PR would be great!

Comment (herbetom) [reply configuration/configuration.py]: So just the following?
```
REMOTE_AUTH_DEFAULT_PERMISSIONS = os.environ.get('REMOTE_AUTH_DEFAULT_PERMISSIONS', '{}')
```
I don't have an idea what would be a good way to build something similar to the way how lists are entered.

([REMOTE_AUTH_DEFAULT_PERMISSIONS](https://github.com/netbox-community/netbox/blob/90dbe9bf60ab3c72be10fd070c65c26dca543ca5/docs/configuration/optional-settings.md#remote_auth_default_permissions) has become [with 2.9](https://github.com/netbox-community/netbox/blob/dde52309d25d6fd6ecc5f05c7a373fa81cb719dc/docs/release-notes/version-2.9.md#configuration-changes) a dictionary.)

Expected inputs would be for example: `{'dcim.add_site': None, 'dcim.change_site': None}`

**Comment by herbetom** [comment]
Since #310 was updated i'm closing this.

**Comment by herbetom** [comment]
Since #310 was updated i'm closing this.

**Pull request closed without merging**